pytest-embedded-idf/pytest_embedded_idf/serial.py

```diff
@@ -207,15 +195,12 @@ def dump_flash(
         else:
             raise ValueError('You must specify "partition" or ("address" and "size") to dump flash')
 
-        content = self.stub.read_flash(_addr, _size)
         if output:
-            if isinstance(output, str):
-                os.makedirs(os.path.dirname(output), exist_ok=True)
-                with open(output, 'wb') as f:
-                    f.write(content)
-            else:
-                output.write(content)
+            esptool.main(['read_flash', str(_addr), str(_size), str(output)], esp=self.esp)
         else:
+            output = 'read_flash.tmp'
+            esptool.main(['read_flash', str(_addr), str(_size), output], esp=self.esp)
+            content = open(output, 'rb').read()
             return content
 
     @EspSerial.use_esptool()
```

Comment:
is `use_esptool()` still useful here? shall we deprecate this function and stop using it in the codebase?

Reply:
I suggest it is still useful because it helps with redirecting the thread.

Reply:
But I'm also interested in the possibility of initializing `self.esp` only once in the `__init__` function: `self.esp = esptool.detect_chip(self.proc, self.baud)`.